File: java/src/televentas/InventarioMemoria.java
package televentas;

import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Optional;

public class InventarioMemoria implements InventarioService {

    private final Map<String, Producto> productos = new HashMap<>();

    public InventarioMemoria(List<Producto> productosIniciales) {
        for (Producto producto : productosIniciales) {
            this.productos.put(producto.getCodigo(), producto);
        }
    }

    @Override
    public Optional<Producto> obtenerProducto(String codigo) {
        return Optional.ofNullable(productos.get(codigo));
    }

    @Override
    public int consultarDisponibilidad(String codigo) {
        Optional<Producto> p = obtenerProducto(codigo);
        if (!p.isPresent()) {
            return 0;
        }
        return p.get().getCantidadDisponible();
    }

    @Override
    public void actualizarStock(String codigo, int cantidad) {
        Producto producto = productos.get(codigo);
        if (producto == null) {
            throw new IllegalArgumentException("Producto con codigo " + codigo + " no encontrado.");
        }
        producto.actualizarStock(cantidad);
    }
}

File: javascript/src/televentas/inventarioMemoria.js
import { InventarioService } from "./inventarioService.js";

export class InventarioMemoria extends InventarioService {
  /**
   * @param {import('./producto.js').Producto[]} productosIniciales
   */
  constructor(productosIniciales) {
    super();
    /** @type {Map<string, import('./producto.js').Producto>} */
    this.productos = new Map();
    for (const producto of productosIniciales) {
      this.productos.set(producto.getCodigo(), producto);
    }
  }

  /** @returns {import('./producto.js').Producto | null} */
  obtenerProducto(codigo) {
    return this.productos.get(codigo) ?? null;
  }

  consultarDisponibilidad(codigo) {
    const p = this.obtenerProducto(codigo);
    if (p == null) {
      return 0;
    }
    return p.getCantidadDisponible();
  }

  actualizarStock(codigo, cantidad) {
    const producto = this.productos.get(codigo);
    if (producto == null) {
      throw new Error(`Producto con codigo ${codigo} no encontrado.`);
    }
    producto.actualizarStock(cantidad);
  }
}
